blocks/bricks/recurrent.py

```diff
@@ -547,43 +527,20 @@ def apply(self, inputs, update_inputs=None, reset_inputs=None,
             Next states of the network.
 
         """
-        if (self.use_update_gate != (update_inputs is not None)) or \
-                (self.use_reset_gate != (reset_inputs is not None)):
-            raise ValueError("Configuration and input mismatch: You should "
-                             "provide inputs for gates if and only if the "
-                             "gates are on.")
-
-        states_reset = states
-
-        if self.use_reset_gate:
-            reset_values = self.gate_activation.apply(
-                states.dot(self.state_to_reset) + reset_inputs)
-            states_reset = states * reset_values
-
+        gate_values = self.gate_activation.apply(
+            states.dot(self.state_to_gates) + gate_inputs)
+        update_values = gate_values[:, :self.dim]
+        reset_values = gate_values[:, self.dim:]
```

Comment:
Did you benchmark if this is faster than two separate matrix multiplications? I can remember @pbrakel saying that in Theano the benefit from performing a single GEMM is actually cancelled out by the slicing operations needed.